### ISSUE TITLE
add_error! generates an unused variable warning

### ISSUE THREAD

**Issue by shepmaster** [Geal/nom#366]
This parser:

```rust
named!(example, add_error!(ErrorKind::Custom(0), tag!("b")));
```

Generates this error:

```
warning: unused variable: `e`, #[warn(unused_variables)] on by default
   --> main.rs:280:5
    |
280 |     named!(example, add_error!(ErrorKind::Custom(0), tag!("b")));
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in a macro outside of the current crate
```

**Comment by Geal** [comment]
that one annoys me a lot, but I have not found a way to fix it yet.

**Comment by sdroege** [comment]
I think this needs some language addition. To be able to mark macro arguments as intentionally unused (which has to somehow propagate to the code after macro evaluation), similar to the underscore prefixed function arguments or variables.

**Comment by Geal** [comment]
I can add `#[allow(unused_variables)]` to the function created with `named!`, it looks like it works. But it is not a satisfying solution

**Comment by sdroege** [comment]
The problem with that is that it will also not warn about unintentionally unused variables, but I guess that's what you mean with not liking it :)

**Comment by Geal** [comment]
I added the function attribute to `named!`, this will fix it for now ( 20d4670 ).
